src/components/button/button.test.tsx

```diff
@@ -0,0 +1,330 @@
+});
+
+it("renders with a 'subtext' element when the 'size' prop is 'large' and text children is passed", () => {
+  render(<Button size="large">foo</Button>);
+
+  // query selector here is the only way we can locate the subtext element, as it has no text or other discernable roles or attributes
+  // eslint-disable-next-line testing-library/no-node-access
+  const subtext = document.querySelector('[data-element="subtext"]');
+  expect(subtext).toBeVisible();
+});
+
+it("does not render, as an invariant is fired due to no text children or 'iconType' prop being defined", () => {
+  const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  expect(() => render(<Button />)).toThrow(
+    "Either prop `iconType` must be defined or this node must have children."
+  );
+
+  consoleSpy.mockRestore();
+});
+
+it("does not render, as an invariant is fired when the 'size' prop is 'small' and the 'subtext' prop is passed", () => {
+  const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  expect(() =>
+    render(
+      <Button subtext="bar" size="small">
+        foo
+      </Button>
+    )
+  ).toThrow("subtext prop has no effect unless the button is large");
+
+  consoleSpy.mockRestore();
+});
+
+it("does not render, as an invariant is fired when the 'size' prop is 'medium' and the 'subtext' prop is passed", () => {
+  const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  expect(() =>
+    render(
+      <Button subtext="bar" size="medium">
+        foo
+      </Button>
+    )
+  ).toThrow("subtext prop has no effect unless the button is large");
+
+  consoleSpy.mockRestore();
+});
+
+describe("props", () => {
+  it("sets the 'aria-label' attribute correctly when a custom value is passed to the 'aria-label' prop", () => {
+    render(<Button aria-label="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("aria-label", "foo");
+  });
+
+  it("sets the 'aria-labelledby' attribute correctly when a custom value is passed to the 'aria-labelledby' prop", () => {
+    render(<Button aria-labelledby="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("aria-labelledby", "foo");
+  });
+
+  it("sets the 'aria-describedby' attribute correctly when a custom value is passed to the 'aria-describedby' prop", () => {
+    render(<Button aria-describedby="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("aria-describedby", "foo");
+  });
+
+  /* Styling test for coverage */
+  it("sets the correct icon colour when 'gradient-white' is passed to the 'buttonType' prop", () => {
+    render(
+      <Button aria-label="button" buttonType="gradient-white" iconType="bin">
+        foo
+      </Button>
+    );
+
+    const icon = screen.getByTestId("icon");
+    expect(icon).toHaveStyle("color: --colorsActionMinorYin090");
```

Comment:
**suggestion/nitpick:** You seem to have used this format elsewhere. Just to keep things consistent in the file.

```suggestion
    const icon = screen.getByTestId("icon");
    
    expect(icon).toHaveStyle("color: --colorsActionMinorYin090");
```